modules/busco/meta.yml

```diff
@@ -0,0 +1,52 @@
+name: busco
+description: Benchmarking Universal Single Copy Orthologs
+keywords:
+  - quality control
+  - genome
+  - transcriptome
+  - proteome
+tools:
+  - busco:
+      description: BUSCO provides measures for quantitative assessment of genome assembly, gene set, and transcriptome completeness based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs selected from OrthoDB.
+      homepage: https://busco.ezlab.org/
+      documentation: ttps://busco.ezlab.org/busco_userguide.html
+      tool_dev_url: https://gitlab.com/ezlab/busco
+      doi: "10.1007/978-1-4939-9173-0_14"
+      licence: ['MIT']
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - fasta:
+      type: file
+      description: Nucleic or amino acid sequence file in FASTA format
+      pattern: "*.{fasta}"
```

Comment:
Are you sure it cannot handle compressed files? It might be worth to add a conditional unzipping to allow zipped input! Because modules should output zipped `fasta` (and therefore allow compressed input), according to the module guidelines.
Also, does it require the fasta extension? If not, remove it or give more choice (`fa,fa.gz`,etc).

Reply:
I just verified that BUSCO crashes on compressed FASTA input.  Following recent discussions on Slack, transparent decompression can be achieved with the following patch.  I chose `__UNCOMPRESSED_FASTA_FILE__` as output name to make it extremely unlikely that it clashes with an existing file.

```
index 6a1a5644..8f3b1249 100644
--- a/modules/busco/main.nf
+++ b/modules/busco/main.nf
@@ -33,11 +33,13 @@ process BUSCO {
     # Copy the image's AUGUSTUS config directory if it was not provided to the module
     [ ! -e augustus_config ] && cp -a /usr/local/config augustus_config
     AUGUSTUS_CONFIG_PATH=augustus_config \\
+    # Ensure the input is uncompressed
+    gzip -cdf $fasta > __UNCOMPRESSED_FASTA_FILE__
     busco \\
         $options.args \\
         --augustus \\
         --cpu $task.cpus \\
-        --in  $fasta \\
+        --in __UNCOMPRESSED_FASTA_FILE__ \\
         --out $meta.id
```

I verified that it works with the following change to the the test file.

```
diff --git a/tests/modules/busco/main.nf b/tests/modules/busco/main.nf
index bf03cf10..1cc8fbdb 100644
--- a/tests/modules/busco/main.nf
+++ b/tests/modules/busco/main.nf
@@ -8,8 +8,7 @@ include { BUSCO } from '../../../modules/busco/main.nf' addParams( options: [arg
 workflow test_busco {
     
        compressed_genome_file = file(params.test_data['bacteroides_fragilis']['genome']['genome_fna_gz'], checkIfExists: true)
-       GUNZIP ( compressed_genome_file )
-       input = GUNZIP.out.gunzip.map { row -> [ [ id:'test' ], row] }
+       input = [ [ id:'test' ], compressed_genome_file ]
```

I also opened an [issue upstream to request transparent decompression of input files](https://gitlab.com/ezlab/busco/-/issues/504).

Reply:
Hm I'd rather thought rather to add in the `script` block about something like 
```
if [[ $file != *.gz ]]; then
    gzip -n $fasta > $fasta.gz
fi
```
(code not tested! I am pretty sure there are examples in existing modules!)